GVFS/GVFS/CommandLine/HealthVerb.cs

```diff
@@ -0,0 +1,246 @@
+{
+    [Verb(HealthVerb.HealthVerbName, HelpText = "Get statistics for the health state of the repository")]
+    public class HealthVerb : GVFSVerb.ForExistingEnlistment
+    {
+        private const string HealthVerbName = "health";
+        private const decimal MaximumHealthyHydration = 0.5m;
+
+        [Option(
+            'n',
+            Required = false,
+            HelpText = "The number of directories to display hydration levels for")]
+        public int DirectoryDisplayCount { get; set; } = 5;
+
+        [Option(
+            'd',
+            "directory",
+            Required = false,
+            HelpText = "Run the statistics tool on a specific directory")]
+        public string Directory { get; set; }
+
+        protected override string VerbName
+        {
+            get { return HealthVerbName; }
+        }
+
+        protected override void Execute(GVFSEnlistment enlistment)
+        {
+            // The path to the root of git's tree
+            string sourceRoot = enlistment.WorkingDirectoryRoot;
+
+            if (this.Directory == null)
+            {
+                this.Directory = string.Empty;
+            }
+            else if (this.Directory.Equals("."))
+            {
+                this.Directory = Environment.CurrentDirectory.Substring(sourceRoot.Length);
+            }
+
+            // Get all of the data needed to calculate statistics then pass them into GVFSEnlistmentStatistics
+            List<string> modifiedPathsFileList = new List<string>();
+            List<string> modifiedPathsFolderList = new List<string>();
+            List<string> placeholderFilePathList = new List<string>();
+            List<string> placeholderFolderPathList = new List<string>();
+
+            this.GetPlaceholdersFromDatabase(enlistment, out List<IPlaceholderData> filePlaceholders, out List<IPlaceholderData> folderPlaceholders);
+            foreach (IPlaceholderData placeholderData in filePlaceholders)
+            {
+                placeholderFilePathList.Add(placeholderData.Path);
+            }
+
+            foreach (IPlaceholderData placeholderData in folderPlaceholders)
+            {
+                placeholderFolderPathList.Add(placeholderData.Path);
+            }
+
+            foreach (string path in this.GetModifiedPathsFromPipe(enlistment))
+            {
+                if (path.Last() == GVFSConstants.GitPathSeparator)
+                {
+                    path.TrimEnd('/');
+                    modifiedPathsFolderList.Add(path);
+                }
+                else
+                {
+                    modifiedPathsFileList.Add(path);
+                }
+            }
+
+            this.GetPathsFromGitIndex(enlistment, out List<string> gitFilePaths, out List<string> gitFolderPaths, out List<string> skipWorkTreeFiles);
+
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentPathData pathData = new GVFSEnlistmentHealthCalculator.GVFSEnlistmentPathData(
+                gitFolderPaths,

```

Comment:
You have incorrect amounts of tabbing in multiple locations, especially with these long parameter lists. Good thing long parameter lists is a code smell that I like to avoid and have already suggested that you do something different.